Documentation using gh-pages (#57)
open-energy-transition/form-energy-storage: documentation/13-kpis.rst

Reviewer: My general feeling is that the page is still a bit cryptic. We probably need some introductions and examples.

##########################################
KPIs Configuration
##########################################

> Plot_KPIs creates both predefined and configurable figures. The predefined figures are:

Reviewer: ```suggestion
`Plot_KPIs` creates both predefined and configurable figures. The predefined figures are:
```

> - A map of curtailed energy for Germany and the wider region.
> - A map of transmission line loading for Germany and the wider region.
> - An overview plot of the energy trade.
> - A summary CSV for storage capacities.

Reviewer: ```suggestion
- a map of curtailed energy for Germany and the wider region,
- a map of transmission line loading for Germany and the wider region,
- an overview plot of the energy trade,
- a summary CSV for storage capacities.
```

The configurable figures are defined based on ``config/config.kpi.yaml``.

Key components of each configuration:

1. ``extract``: Defines the network statistics for the figures (e.g., system cost, generation, storage, emissions).
2. ``include`` and/or ``exclude``: Specifies the regions or entities to include or exclude in the calculation (e.g., "DE" for Germany, "EU" for the European Union).
3. ``carrier_filter``: Specifies the energy carriers or technologies (e.g., power, electricity, storage, heat) relevant to the figure.
4. ``group_carrier``: Specifies the names used for each carrier. Same names are aggregated.
5. ``plot``: Determines the type of plot or visualization to be used (e.g., "detail" for detailed data, "overview" for broader data representation).
6. ``figsize``: Determines the figsize of the plots. If not defined, the default size are chosen.
7. ``plot_kw``: Additional keyword arguments for the plot (e.g., title, labels, and axis).

> Breakdown of the components:

Reviewer: This should be a title and an introduction.

``extract``
------------------------

> - **system cost**: Extract data from ``csvs/nodal_costs.csv``.
> - **capacity**: Extract data from ``csvs/nodal_capacities.csv``.
> - **capacity stats**: Extract capacity data from ``n.statistics``. See notes for more detail.
> - **generation**: Extract power generation data from ``n.statistics.energy_balance``.
> - **emission**: Extract emissions data based on emission links to the atmosphere.
> - **energy balance**: Extract energy balance data from ``n.statistics.energy_balance``.
> - **SOC**: Extract state of charge from the ``storage_units`` component.

Reviewer: ```suggestion
- **system cost**: Extract data from ``csvs/nodal_costs.csv``.
- **capacity**: Extract data from ``csvs/nodal_capacities.csv``.
- **capacity stats**: Extract capacity data from ``n.statistics``. See the note below for more detail.
- **generation**: Extract power generation data from ``n.statistics.energy_balance()``.
- **emission**: Extract emissions data based on emission links to the atmosphere.
- **energy balance**: Extract energy balance data from ``n.statistics.energy_balance()``.
- **SOC**: Extract state of charge from the ``storage_units`` component.
```

It's a bit counterintuitive that you need to add `stats` to use `n.statistics` for capacities, but not for the next ones.

.. note::
    **capacity stats** is a special option because it has two extra components:

>     ``stats``: Three options:

Reviewer: Create a sentence

>     - **install**: Extract capacity data from ``n.statistics.installed_capacity``
>     - **optimal**: Extract capacity data from ``n.statistics.optimal_capacity``

Reviewer: ```suggestion
    - **install**: Extract capacity data from ``n.statistics.installed_capacity()``
    - **optimal**: Extract capacity data from ``n.statistics.optimal_capacity()``
```

>     - **expand**:  The difference between optimal minus install.

Reviewer: Are you computing it yourself ? Ref https://pypsa.readthedocs.io/en/latest/api/_source/pypsa.statistics.StatisticsAccessor.expanded_capacity.html#pypsa.statistics.StatisticsAccessor.expanded_capacity

    ``storage``: If set to ``true``, only storage capacities of the component ``store`` and ``storage_units`` are taken into account.

``carrier_filter``
------------------------

> a. for system cost, capacity, capacity stats, generation, and emission:

Reviewer: This needs to be a title

- **electricity**: Filter carrier with AC bus carrier.
- **electricity+**: Filter carrier with AC bus carrier and also include water tanks and EV batteries.
- **storage**: Filter carrier with all storage-related technologies (``storage_units``, ``links``, and ``stores``).
- **storage-cap**: Filter carrier with all storage capacity-related technologies (``storage_units`` and ``links``).
- **storage-energy**: Filter carrier with all storage energy-related technologies (``storage_units`` and ``stores``).
- **power**: Filter carrier with power generation technologies.

.. note::
    **storage**, **storage-cap**, **storage-energy**, **power** is based on a list in ``config/config.kpi.yaml``.
    You can create your own filtering scheme by adding a list in ``kpi: filter_scheme``

    .. literalinclude:: ../config/config.kpi.yaml
       :language: yaml
       :start-at: power:
       :end-at: ]

> b. for energy balance:

Reviewer: This needs to be a title

##########################################
KPIs Configuration
##########################################

`Plot_KPIs` creates both predefined and configurable figures. The predefined figures are:

* A map of curtailed energy for Germany and the wider region.
* A map of transmission line loading for Germany and the wider region.
* An overview plot of the energy trade.
* A summary CSV for storage capacities.

The configurable figures are defined based on ``config/config.kpi.yaml``.

.. note::

>    The PyPSA-Eur configuration follows a pyramid-like structure, where the parameters in the highest configuration override those in the configurations below it. 

Reviewer 2: ```suggestion
   The PyPSA-Eur configuration files follow a pyramid-like structure, where the parameters in the highest configuration file add to and override those in the configuration file below it. 
```

   The order is as follows:

   1. `scenarios.form.yaml <https://open-energy-transition.github.io/form-energy-storage/12-scenarios.html>`_
   2. `config.form.yaml <https://open-energy-transition.github.io/form-energy-storage/11-baseline.html>`_
   3. `config.default.yaml <https://pypsa-eur.readthedocs.io/en/latest/configuration.html>`_
   4. config.kpi.yaml (**this section**)

> 

Reviewer 2: ```suggestion

Thus, for example changes specified in `scenario.form.yaml` will add to and override configurations in `config.form.yaml` and so on.

```

4. ``group_carrier``: Specifies the names used for each carrier. Same names are aggregated.
5. ``plot``: Determines the type of plot or visualization to be used (e.g., "detail" for detailed data, "overview" for broader data representation).
6. ``figsize``: Determines the figsize of the plots. If not defined, the default size are chosen.
7. ``plot_kw``: Additional keyword arguments for the plot (e.g., title, labels, and axis).

Components
---------------------------

``extract``
==========================

* **system cost**: Extract data from ``csvs/nodal_costs.csv``.
* **capacity**: Extract data from ``csvs/nodal_capacities.csv``.
* **capacity stats**: Extract capacity data from ``n.statistics``. See notes for more detail.
* **generation**: Extract power generation data from ``n.statistics.energy_balance``.
* **emission**: Extract emissions data based on emission links to the atmosphere.
* **energy balance**: Extract energy balance data from ``n.statistics.energy_balance``.
* **SOC**: Extract state of charge from the ``storage_units`` component.

.. note::
    **capacity stats** is a special option because it has two extra components:

>     ``stats``: Three options:

Author: ```suggestion
    ``stats``: Extract capacity data from these three options:
```

    
    * **install**: Extract capacity data from ``n.statistics.installed_capacity``
    * **optimal**: (Default) Extract capacity data from ``n.statistics.optimal_capacity``
    * **expand**:  The difference between optimal minus install.

    ``storage``: If set to ``true``, only storage capacities of the component ``store`` and ``storage_units`` are taken into account.

``carrier_filter``
==========================

> a. for system cost, capacity, capacity stats, generation, and emission:

Author: ```suggestion
**a. for system cost, capacity, capacity stats, generation, and emission**:
```

* **electricity**: Filter carrier with AC bus carrier.
* **electricity+**: Filter carrier with AC bus carrier and also include water tanks and EV batteries.
* **storage**: Filter carrier with all storage-related technologies (``storage_units``, ``links``, and ``stores``).
* **storage-cap**: Filter carrier with all storage capacity-related technologies (``storage_units`` and ``links``).
* **storage-energy**: Filter carrier with all storage energy-related technologies (``storage_units`` and ``stores``).
* **power**: Filter carrier with power generation technologies.

.. note::
    **storage**, **storage-cap**, **storage-energy**, **power** is based on a list in ``config/config.kpi.yaml``.
    You can create your own filtering scheme by adding a list in ``kpi: filter_scheme``

    .. literalinclude:: ../config/config.kpi.yaml
       :language: yaml
       :start-at: power:
       :end-at: ]

> b. for energy balance:

Author: ```suggestion
**b. for energy balance**:
```

* **electricity**: Filter carrier with AC bus carrier.
* **electricity+**: Filter carrier with AC bus carrier and also include solar rooftops, BEV chargers, and Vehicle-to-Gas.
* **low voltage**: Filter carrier with low voltage bus carrier.
* **storage**: Filter carrier with all storage capacity-related technologies.
* **heat**: Filter carrier with heat bus carrier.
* **hydrogen**: Filter carrier with hydrogen bus carrier.

> c. for SOC can be defined individually or in arrays of ``storage_units`` carriers.

Author: ```suggestion
**c. for SOC**, this can be defined individually or in arrays of ``storage_units`` carriers.
```
